auth/auth.go

```diff
@@ -101,6 +108,23 @@ func (a ACL) VerifyPeerCertificateServer(rawCerts [][]byte, verifiedChains [][]*
 		return nil
 	}
 
+	// Check against OPA
+	if a.AllowOPAQuery != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		input := map[string]interface{}{
+			"certificate": cert,
+		}
+		results, err := a.AllowOPAQuery.Eval(ctx, rego.EvalInput(input))
+		if err != nil {
+			// TODO logging? Or?
+			return err
```

Comment:
Let's just return the error here, no logging. But we should wrap it:
```
return fmt.Errorf("unauthorized: policy returned error: %w", err)
```

Same thing below for client